frontend/src/Editor/QueryManager/QueryManager.jsx

```diff
@@ -418,22 +369,26 @@ let QueryManager = class QueryManager extends React.Component {
         {(addingQuery || editingQuery) && (
           <div className="py-2">
             {currentTab === 1 && (
-              <div className="row row-deck px-2 pt-1 query-details">
+              <div className="row row-deck px-2 mt-0 query-details">
                 {dataSources && mode === 'create' && (
-                  <div className="datasource-picker mb-2">
+                  <div className="datasource-picker mt-1 mb-2">
                     <label className="form-label col-md-2">Datasource</label>
-                    <Select
+                    <SelectSearch
                       options={[
                         ...dataSources.map((source) => {
-                          return { label: source.name, value: source.id };
+                          return { name: source.name, value: source.id, kind: source.kind };
                         }),
                         ...staticDataSources.map((source) => {
-                          return { label: source.name, value: source.id };
+                          return { name: source.name, value: source.id, kind: source.kind };
                         }),
                       ]}
-                      onChange={(newValue) => this.changeDataSource(newValue.value)}
+                      value={selectedDataSource ? selectedDataSource.id : ''}
+                      search={true}
+                      onChange={(value) => this.changeDataSource(value)}
+                      filterOptions={fuzzySearch}
+                      renderOption={this.renderDataSourceOption}
                       placeholder="Select a data source"
-                      styles={selectStyles}
```

Comment:
I thikn `selectStyles` is required for some dark mode fixes

Reply:
okay, for `react-select` we need selectStyles. resolving this